### PR TITLE
Implemented Login

### PR DESCRIPTION
**Pull request by matteofiordarancio**

### CODE REVIEW

**Comment by GiovanniGobbs** [comment]
Why there is a debug.keystore file tracked?

File: app/src/main/java/com/uncookthebook/app/GoogleActivity.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.uncookthebook.app;
+
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+
+public interface GoogleActivity {
+    GoogleSignInClient getGoogleClient();
+
+    GoogleSignInAccount getGoogleAccount();
+    void setGoogleAccount(GoogleSignInAccount account);
```
Comment (GiovanniGobbs) [inline app/src/main/java/com/uncookthebook/app/GoogleActivity.java:13]: maybe add some documentation to understand what the interface models and its methods do.

File: app/src/main/java/com/uncookthebook/app/LoginFragment.java [hunk 2]
```diff
@@ -1,59 +1,76 @@
...
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
 import com.google.android.material.button.MaterialButton;
-import com.google.android.material.textfield.TextInputEditText;
-import com.google.android.material.textfield.TextInputLayout;
+
+import java.util.Objects;
 
 /**
  * Fragment representing the login screen for Shrine.
```
Comment (GiovanniGobbs) [inline app/src/main/java/com/uncookthebook/app/LoginFragment.java]: Remove references to Shrine

File: app/src/main/java/com/uncookthebook/app/LoginFragment.java [hunk 3]
```diff
@@ -1,59 +1,76 @@
...
  */
 public class LoginFragment extends Fragment {
+    private static final int RC_SIGN_IN = 2;
```
Comment (GiovanniGobbs) [inline app/src/main/java/com/uncookthebook/app/LoginFragment.java:29]: What is this? Add a comment that tells it

**Comment by matteofiordarancio** [comment]
Because otherwise the app won't work when you try it. Remember last time when you did it it didn't work because of that?

File: app/src/main/java/com/uncookthebook/app/LoginFragment.java [hunk 4]
```diff
@@ -1,59 +1,76 @@
...
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        // Result returned from launching the Intent from GoogleSignInClient.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            // The Task returned from this call is always completed, no need to attach
+            // a listener.
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            handleSignInResult(task);
+        }
```
Comment (GiovanniGobbs) [inline app/src/main/java/com/uncookthebook/app/LoginFragment.java:59]: What if the code is not sign_in? What happens? It remains on this fragment without doing nothing?

**Review by GiovanniGobbs — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/uncookthebook/app/LoginFragment.java (hunk 4 above)
Comment (matteofiordarancio) [reply app/src/main/java/com/uncookthebook/app/LoginFragment.java:59]: check the code in handleSignInResult

**Review by GiovanniGobbs — APPROVED** [review]

**Pull request merged**